pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,198 @@
+package latencyprofilecontroller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	apiconfigv1 "github.com/openshift/api/config/v1"
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	listerv1 "github.com/openshift/client-go/config/listers/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	controlplanev1 "github.com/openshift/api/kubecontrolplane/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+const (
+	apiServerConfigMapName = "config"
+	apiServerConfigMapKey  = "config.yaml"
+
+	DefaultNotReadyTolerationSecondsArgument    = "default-not-ready-toleration-seconds"
+	DefaultUnreachableTolerationSecondsArgument = "default-unreachable-toleration-seconds"
+)
+
+// LatencyProfileController periodically lists the config/v1/node object
+// and fetches the worker latency profile applied on the cluster which is used to
+// updates the status of the config node object. The status updates reflect the
+// state of kube-api-server(s) running on control plane node(s) and their
+// observed config for default-not-ready-toleration-seconds and
+// default-unreachable-toleration-seconds match the applied arguments.
+
+type latencyProfileController struct {
+	operatorClient   v1helpers.StaticPodOperatorClient
+	configClient     configv1.ConfigV1Interface
+	configMapClient  corev1client.ConfigMapsGetter
+	configNodeLister listerv1.NodeLister
+}
+
+func NewLatencyProfileController(
```

Comment:
if you decide to set the status for this operator (https://github.com/openshift/cluster-kube-apiserver-operator/pull/1328#discussion_r857344419) then we will need a similar controller for `kube-controller-manager` and `kube-scheduler`. We could write a generic controller and reuse it.

Reply:
https://github.com/openshift/cluster-kube-apiserver-operator/pull/1328#discussion_r871977546